toolkit/offchain/src/collateral_selection.rs

```diff
@@ -0,0 +1,77 @@
+use ogmios_client::types::*;
+
+pub const MIN_REQUIRED_COLLATERAL: u64 = 5_000_000;
+
+// Largest-First selection algorithm based on https://cips.cardano.org/cip/CIP-0002#largest-first
+pub fn largest_first(
+	inputs_available: &[OgmiosUtxo],
+	amount_required: u64,
+	max_collateral_inputs: u32,
+	min_required_collateral: u64,
+) -> Result<Vec<&OgmiosUtxo>, String> {
+	let amount_required = amount_required.max(min_required_collateral);
+	let amount_available: u64 = inputs_available.iter().map(|utxo| utxo.value.lovelace).sum();
+	if amount_available < amount_required {
+		return Err(format!("The available amount ({amount_available}) is less than the required amount ({amount_required})"));
```

Comment:
In this case we should handle this error by adding the context of collateral selection, which is IMO redundant, because it is returned from the `collateral_selection` module.

I mean that if we display this message to the user, he will know available amount of something is not less that required but he won't know it's about collateral.

Reply:
How about `The available available amount of lovelace (110) is less than the required collateral amount (111)`?